src/broadcast.jl

```diff
@@ -25,7 +25,7 @@ if VERSION < v"0.6.0-dev" # Old approach needed for inference to work
     @inline ziptype(A, B, C, D...) = Zip{Tuple{eltype(A)}, ziptype(B, C, D...)}
 
     nullable_broadcast_eltype(f, As...) =
-        _default_eltype(Base.Generator{ziptype(As...), ftype(f, As...)})
+        eltype(_default_eltype(Base.Generator{ziptype(As...), ftype(f, As...)}))
```

Comment:
But, wouldn't that warrants that you don't get a `Nullable{T}` here?

Given that `eltype(NullableArray([1], [true])) == Int`, that implies `_default_eltype(+, NullableArray([1], [true]), NullableArray([1], [true])) == Int`, right?